src/api/common/utils/CommonCalendarUtils.ts

```diff
@@ -90,10 +90,27 @@ export function getEventElementMinId(timestamp: number): string {
 	return createEventElementId(timestamp, -DAYS_SHIFTED_MS)
 }
 
+/**
+ * return a cleaned and comparable version of a mail address without leading/trailing whitespace or uppercase characters.
+ */
+export function cleanMailAddress(address: string): string {
+	return address.trim().toLowerCase()
+}
+
 /**
  * get the first attendee from the list of attendees/guests that corresponds to one of the given recipient addresses, if there is one
  */
 export function findAttendeeInAddresses<T extends { address: EncryptedMailAddress }>(attendees: ReadonlyArray<T>, addresses: ReadonlyArray<string>): T | null {
-	const lowerCaseAddresses = addresses.map((r) => r.toLowerCase())
-	return attendees.find((a) => lowerCaseAddresses.includes(a.address.address.toLowerCase())) ?? null
+	// the filters are necessary because of #5147
+	// we may get passed addresses and attendees that could not be decrypted and don't have addresses.
+	const lowerCaseAddresses = addresses.filter(Boolean).map(cleanMailAddress)
+	return attendees.filter((a) => a.address.address != null).find((a) => lowerCaseAddresses.includes(cleanMailAddress(a.address.address))) ?? null
```

Comment:
does `filter()` make sense here or can we just do `find()` straight away.

Reply:
true